processing/src/main/java/org/apache/druid/segment/filter/OrFilter.java

```diff
@@ -71,6 +76,167 @@ public OrFilter(List<Filter> filters)
     this(new LinkedHashSet<>(filters));
   }
 
+  @Override
+  public <T> FilterBundle makeFilterBundle(
+      ColumnIndexSelector columnIndexSelector,
+      BitmapResultFactory<T> bitmapResultFactory,
+      int selectionRowCount,
+      int totalRowCount,
+      boolean includeUnknown,
+      boolean allowPartialIndex
+  )
```

Comment:
I find the logic in this method a little bit hard to follow.  I think it would help to have a bit of a pseudo-code explanation of how the method is trying to do its job.

The pseudo-code that I can imagine for this method (but which I'm pretty sure is currently not implemented by this method) looks like

1) Compute all sub-filter bundles, store them all independently in a list
2) If any of the sub-filter bundles requires a matcher, then convert all bitmap indexes into bitmap index matchers that compare rowIds against the bitmap index and return a matcher-only bundle.
3) If all sub-filters are indexes, then OR them together and return an index-only bundle

This means that an OR filter will never produce a "partial bundle".  It will either produce an index or a matcher bundle, but not both.

Reply:
it is more or less doing the pseudo-code you listed here, will try to update the comments to clear things up